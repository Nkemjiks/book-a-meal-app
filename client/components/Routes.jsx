import React from 'react';
import { Route, Switch, withRouter } from 'react-router-dom';
import Navbar from './Navbar';
import ManageMeal from './Caterer/ManageMeal';
import ManageMenu from './Caterer/ManageMenu';
import ManageCatererOrder from './Caterer/ManageOrder';
import Dashboard from './Customer/Dashboard';
import ManageCustomerOrder from './Customer/ManageOrder';
import Login from './Login';
import Signup from './Signup';
import LandingPage from './LandingPage';
import { UserProtectComponent, RoleProtectedComponent } from './Protect';

/**
 * route for each component
 *
 * @param {none}
 *
 * @returns {JSX} JSX representation of component
 */
const Routes = () => (
  <div>
    <Navbar />
    <Switch>
      <Route exact path="/" component={LandingPage} />
      <Route path="/signup" component={Signup} />
      <Route path="/login" component={Login} />
      <UserProtectComponent path="/customer/dashboard" component={Dashboard} />
      <UserProtectComponent path="/customer/order" component={ManageCustomerOrder} />
      <RoleProtectedComponent path="/caterer/meal" component={ManageMeal} />
      <RoleProtectedComponent path="/caterer/menu" component={ManageMenu} />
      <RoleProtectedComponent path="/caterer/order" component={ManageCatererOrder} />
    </Switch>
  </div>
);

export default withRouter(Routes);
