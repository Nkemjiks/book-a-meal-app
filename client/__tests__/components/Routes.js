import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Routes from '../../components/Routes';

Enzyme.configure({ adapter: new Adapter() });

const wrapper = shallow(<Routes />);

describe('Container Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders page Route', () => {
    expect((wrapper).find('Provider').length).toBe(1);
    expect((wrapper).find('Router').length).toBe(1);
    expect((wrapper).find('Switch').length).toBe(1);
    expect((wrapper).find('Route').length).toBe(4);
    expect((wrapper).find('UserProtected').length).toBe(2);
    expect((wrapper).find('RoleProtected').length).toBe(3);
  });
});
