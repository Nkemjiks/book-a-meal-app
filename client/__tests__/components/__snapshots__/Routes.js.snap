// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component renders correctly 1`] = `
<div>
  <div
    id="nav"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        src="image/logo.png"
      />
    </a>
  </div>
  <div
    className="center landing"
  >
    <h1
      id="h1"
    >
      404
    </h1>
    <h2
      id="h2"
    >
      Page Not Found
    </h2>
  </div>
  <div />
</div>
`;
